src/simsopt/geo/magneticfieldclasses.py

```diff
@@ -141,10 +142,6 @@ def compute(self, points, compute_derivatives=0):
         if compute_derivatives >= 1:
             self._dB_by_dX = np.array(self.dBlambdify_by_dX(r, z, phi)).transpose((2, 0, 1))
 
-        if compute_derivatives >= 2:
```

Comment:
Here, you have two if conditions and an asset at top. 
While the first if condition looks fine, the second one is checking for compute_derivatives >= 2, where as the assert condition is checking for compute_derivatives <= 2. 

The current code works, but it is better to tighten it to prevent future bugs.

Reply:
Wouldn't that be fixed with the latest commit where the check for compute_derivatives >= 2 is not made?